src/embind/embind.js

```diff
@@ -167,18 +167,23 @@ var LibraryEmbind = {
     return errorClass;
   },
 
-
-  $createNamedFunction__deps: ['$makeLegalFunctionName'],
   $createNamedFunction: function(name, body) {
-    name = makeLegalFunctionName(name);
-    // Use an abject with a computed property name to create a new function with
-    // a name specified at runtime, but without using `new Function` or `eval`.
-    return {
-      [name]: function() {
-        return body.apply(this, arguments);
-      }
-    }[name];
+    return Object.defineProperty(body, 'name', {
+      value: name
+    });
```

Comment:
Can you do this whole thing with an arrow function. skipping the `return` and the outer curly braces?

Reply:
Does the old version of this function return a new object while the new version mutates the function in place, is that right/ok?

Reply:
As far as I can tell from usages, it should be fine - we only use it with newly created function which is why we want to have a unique name.

Reply:
> Can you do this whole thing with an arrow function. skipping the return and the outer curly braces?

Hm arrow functions for libraries seem somewhat rare in the codebase generally. If it's something we want more of, maybe worth running a separate codemod step over all libs?

Reply:
Changed this one to arrow function for now.

Reply:
They should be fairly common these days.  I've been converting them over time, where possible.

Reply:
Yeah I just meant we should probably do a single codemod run (see my ast-grep comment in the other PR).